docs/examples/io_tests/test_cases/cpp/setup/compile.sh

```diff
@@ -13,19 +13,19 @@
 unset compilationStatus
 
 #language specific compile and run of each test case
-g++ -std=c++11 *.cpp 2>&1 | tee $testLog > /dev/null
-compilationStatus=${PIPESTATUS[0]}
+g++-6 -std=c++11 ./*.cpp 2>&1 | tee "$testLog" > /dev/null
+COMPILATION_STATUS="${PIPESTATUS[0]}"
 
 #collect the log of this compilation to overall log.txt
-cat $testLog >> $log
+cat "$testLog" >> "$log"
 
 #empty log file
 #truncate -s 0 $testLog		#this line gives problem on MAC machines
-rm $testLog
-touch $testLog
+rm "$testLog"
+touch "$testLog"
 
 #exclude any warnings, type cast messages etc and check for compilation errors
-noOfErrors=$(grep -v "^Note:" $testLog | wc -l | awk '{print $1}')
+NO_OF_ERRORS="$(grep -vc "^Note:" "$testLog" awk "{print $1}")"
 
```

Comment:
you can put a shellcheck ignore before this line. The ignore statement is:
\# shellcheck disable=SC2034